ui.apps/src/main/content/jcr_root/apps/etoolbox-rollout-manager/clientlibs/rollout-manager-ui/js/console-ui.dialog.js

```diff
@@ -59,11 +64,37 @@
         dialog.footer.appendChild(closeBtn);
     }
 
-    function insertLogItem(dialog, message, safe) {
-        const logItem = document.createElement('div');
-        logItem.className = 'rollout-manager-log-item';
-        logItem[safe ? 'textContent' : 'innerHTML'] = message;
-        dialog.content.insertAdjacentElement('beforeend', logItem);
+    function updateLog(dialog, message) {
+        if (message.type !== 'rollout' && message.type !== 'activation') return;
+        const rolloutItems = $(dialog).find('.rollout-log-item');
+        const itemToUpdate = rolloutItems.filter((i, item) => item.value === message.path);
+
+        if (message.type === 'rollout') {
+            itemToUpdate.prop('checked', message.result === 'success');
+            itemToUpdate.toggleClass('rollout-log-item-error', message.result === 'error');
+        }
+
+        if (message.type === 'activation') {
+            if (itemToUpdate.find('.rollout-activation-status').length) return;
+            $(`<i class="rollout-activation-status ${message.result === 'error' ? 'error' : ''}">`).text(`${message.result === 'success' ? PUBLISH_SUCCESS_MSG : PUBLISH_ERROR_MSG}`).appendTo(itemToUpdate);
+        }
+    }
```

Comment:
It can be made safer and more readable.
```suggestion
    switch (type) {
        case 'rollout':
            handleRollout(itemToUpdate, result);
            break;
            
        case 'activation':
            handleActivation(itemToUpdate, result);
            break;
    }
}

function handleRollout(item, result) {
    item.prop('checked', result === 'success');
    item.toggleClass('rollout-log-item-error', result === 'error');
}

function handleActivation(item, result) {
    if (item.find('.rollout-activation-status').length) return;
    
    const isError = result === 'error';
    const message = isError ? PUBLISH_ERROR_MSG : PUBLISH_SUCCESS_MSG;
    
    $('<i>')
        .addClass('rollout-activation-status')
        .toggleClass('error', isError)
        .text(message)
        .appendTo(item);
}
```